IT Test: Module (FIM) - Replace callback_generator to generate_monitoring_callback

CamiRomero opened the pull request:
|Related issue|
|---|
[2345](https://github.com/wazuh/wazuh-qa/issues/2534)

<!--
This template reflects sections that must be included in new Pull requests.
Contributions from the community are really appreciated. If this is the case, please add the
"contribution" to properly track the Pull Request.

Please fill the table above. Feel free to extend it at your convenience.
-->

## Description

<!--
Add a clear description of how the problem has been solved.
-->
When PR 2345 was merged, we replace the name of a function callback_generator for generate_monitoring_callback. But some tests were affected by this change.

test affected:

    test_fim/test_files/test_report_changes/test_diff_size_limit_configured.py

    test_fim/test_files/test_report_changes/test_diff_size_limit_default.py

    test_fim/test_registry/test_registry_report_changes/test_registry_file_size_default.py

    test_fim/test_synchronization/test_registry_responses_win32.py

    test_fim/test_synchronization/test_sync_disabled_win32.py

    test_fim/test_synchronization/test_sync_registry_disabled_win32.py

    test_fim/test_synchronization/test_sync_registry_enabled_win32.py

Changes to do:
Replace calbback_generator with generate_monitoring_callback


<!--
When proceed, this section should include new configuration parameters.
-->

## Logs example

<!--
Paste here related logs and alerts
-->
### test_fim/

| Test path | Execution |OS  | Type  | Version  | Date | Status | Executed by | 
|---   |---    |---    |--- |--- |--- |--- |---
| test_fim/ | Jenkins | Windows   | Agent  | 4.3.0 | 01/02/2022 | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19695/) | Camila |
| test_fim/ | Jenkins | Windows  | Agent | 4.3.0 | 01/02/2022 | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19706/) | Camila |
| test_fim/ | Jenkins| Windows   | Agent | 4.3.0 | 01/02/2022| [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19707/) | Camila |

| Test path | Execution |OS  | Type  | Version  | Date | Status | Executed by | 
|---   |---    |---    |--- |--- |--- |--- |---
| test_fim/ | Jenkins | MacOS   | Agent  | 4.3.0 | 01/02/2022 | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19693/) | Camila |
| test_fim/ | Jenkins | MacOS  | Agent | 4.3.0 | 01/02/2022 | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19702/) | Camila |
| test_fim/ | Jenkins| MacOS   | Agent | 4.3.0 | 01/02/2022| [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19703/) | Camila |

| Test path | Execution |OS  | Type  | Version  | Date | Status | Executed by | 
|---   |---    |---    |--- |--- |--- |--- |---
| test_fim/ | Jenkins | Centos   | Agent  | 4.3.0 | 01/02/2022 | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19700/) | Camila |
| test_fim/ | Jenkins | Centos  | Agent | 4.3.0 | 01/02/2022 | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19701/) | Camila |
| test_fim/ | Jenkins| Centos   | Agent | 4.3.0 | 01/02/2022| [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19718/) | Camila |

| Test path | Execution |OS  | Type  | Version  | Date | Status | Executed by | 
|---   |---    |---    |--- |--- |--- |--- |---
| test_fim/ | Jenkins | Centos   | Manager  | 4.3.0 | 01/02/2022 | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19694/) | Camila |
| test_fim/ | Jenkins | Centos  | Manager | 4.3.0 | 01/02/2022 | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19704/) | Camila |
| test_fim/ | Jenkins| Centos   | Manager | 4.3.0 | 01/02/2022| [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19705/) | Camila |

| Test path | Execution |OS  | Type  | Version  | Date | Status | Executed by | 
|---   |---    |---    |--- |--- |--- |--- |---
| test_fim/ | Jenkins | Solaris   | Agente  | 4.3.0 | 02/02/2022 | [:green_circle:  ](https://ci.wazuh.info/view/Tests/job/Test_integration/19820/) | Camila |
| test_fim/ | Jenkins | Solaris  | Agente | 4.3.0 | 02/02/2022 | [:green_circle: ](https://ci.wazuh.info/view/Tests/job/Test_integration/19821/) | Camila |
| test_fim/ | Jenkins| Solaris   | Agente | 4.3.0 | 02/02/2022| [:green_circle: ](https://ci.wazuh.info/view/Tests/job/Test_integration/19822/) | Camila |

### test affected - local
| Test path | Execution |OS  | Type  | Version  | Date | Status | Executed by | 
|---   |---    |---    |--- |--- |--- |--- |---
| test affected | Local | Centos   | Agente  | 4.3.0 | 02/02/2022 | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/7988926/R1.zip)| Camila |
| test affected| Local | Centos  | Agente | 4.3.0 | 02/02/2022 | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/7986668/R2.zip)| Camila |
| test affected | Local| Centos   | Agente | 4.3.0 | 02/02/2022| [:green_circle:](https://github.com/wazuh/wazuh-qa/files/7986669/R3.zip) | Camila |
## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.
<!--
Important: Don't remove these checks if your PR modifies Python code.
-->
- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.

damarisg reviewed (commented):
Could you add the changelog, please? 
Also, Could you attach againt the fisrt Execution local, please? I can't see the results.

damarisg reviewed (approved):
This PR is an example that we need to pay attention to on each merge. 

I attached the list of Test Executions:

| Test path | Execution |OS  | Type  | Version  | Date | Status | Executed by | 
|---   |---    |---    |--- |--- |--- |--- |---
| test_fim/ | Jenkins | Windows   | Agent  | 4.3.0 | 02/02/2022 | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19719) | Seyla |
| test_fim/ | Jenkins | centOS| Agent | 4.3.0 | 02/02/2022 | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19824/) | Seyla |
| test_fim/ | Jenkins| macOS| Agent | 4.3.0 | 02/02/2022| [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19720) | Seyla |
| test_fim/ | Jenkins|  centOS| Manager | 4.3.0 | 02/02/2022| [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19722) | Seyla |
| test_fim/ | Jenkins|  Solaris | Agent | 4.3.0 | 02/02/2022| [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/19823) | Seyla |

CamiRomero commented:
@damarisg I added the Changelog in commit [cfb29](https://github.com/wazuh/wazuh-qa/pull/2535/commits/cfb29b79bfc9239d0d11a379593c5ba00537cbb5). Thanks for noting that the first execution was broke, I updated the executions.
